app/services/azure_notifications_service.rb

```diff
@@ -0,0 +1,55 @@
+class AzureNotificationsService
+  def notify(tags, data, collapse_key: nil)
+    tags = tags.join(' || ') if tags.instance_of?(Array)
+    headers = {
+      'ServiceBusNotification-Format' => 'gcm',
+      'ServiceBusNotification-Tags' => tags,
+      'Content-Type' => 'application/json;charset=utf-8'
+    }
+    body = { data: data }
+    body[:collapse_key] = collapse_key unless collapse_key.nil?
+    send :post, 'messages', body: body.to_json, headers: headers
+  end
+
+  def register_device(handle, tags)
+    res = send :get, "registrations?$filter=#{URI::encode("GcmRegistrationId eq '#{handle}'")}"
+    Nokogiri::XML(res.decoded).css('entry title').each {|n| send :delete, "registrations/#{n.content}", headers: {'If-Match'=>'*'}}
```

Comment:
Line is too long. [131/80]<br>Surrounding space missing for operator '=>'.<br>Space between { and | missing.<br>Space missing inside }.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.